src/components/stegvisning/AFP/AFP.tsx

```diff
@@ -74,122 +101,140 @@ export function AFP({ isLastStep, afp, onCancel, onPrevious, onNext }: Props) {
     setShowAlert(value)
     setValidationError('')
+    if (harSamboer) {
+      setIsLastStep(
+        !(
+          ekskludertStatus?.aarsak === 'HAR_LOEPENDE_UFOERETRYGD' &&
+          value !== 'nei'
+        )
+      )
+    }
   }
 
   return (
-    <Card hasLargePadding hasMargin>
-      <form onSubmit={onSubmit}>
-        <Heading level="2" size="medium" spacing>
-          <FormattedMessage id="stegvisning.afp.title" />
-        </Heading>
-        <BodyLong size="large">
-          <FormattedMessage id="stegvisning.afp.ingress" />
-        </BodyLong>
-        <ReadMore
-          name="Avtalefestet pensjon i offentlig sektor"
-          className={styles.readmoreOffentlig}
-          header={
-            <FormattedMessage id="stegvisning.afp.readmore_offentlig_title" />
-          }
-        >
-          <FormattedMessage id="stegvisning.afp.readmore_offentlig_list_title" />
-          <ul className={styles.list}>
-            <li>
-              <FormattedMessage id="stegvisning.afp.readmore_offentlig_list_item1" />
-            </li>
-            <li>
-              <FormattedMessage id="stegvisning.afp.readmore_offentlig_list_item2" />
-            </li>
-            <li>
-              <FormattedMessage id="stegvisning.afp.readmore_offentlig_list_item3" />
-            </li>
-          </ul>
-          <FormattedMessage id="stegvisning.afp.readmore_offentlig_ingress" />
-        </ReadMore>
-        <ReadMore
-          name="Avtalefestet pensjon i privat sektor"
-          className={styles.readmorePrivat}
-          header={
-            <FormattedMessage id="stegvisning.afp.readmore_privat_title" />
-          }
-        >
-          <FormattedMessage id="stegvisning.afp.readmore_privat_list_title" />
-          <ul className={styles.list}>
-            <li>
-              <FormattedMessage id="stegvisning.afp.readmore_privat_list_item1" />
-            </li>
-            <li>
-              <FormattedMessage id="stegvisning.afp.readmore_privat_list_item2" />
-            </li>
-            <li>
-              <FormattedMessage id="stegvisning.afp.readmore_privat_list_item3" />
-            </li>
-          </ul>
-          <FormattedMessage
-            id="stegvisning.afp.readmore_privat_link"
-            values={{
-              ...getFormatMessageValues(intl),
-            }}
-          />
-        </ReadMore>
-        <RadioGroup
-          className={styles.radiogroup}
-          legend={<FormattedMessage id="stegvisning.afp.radio_label" />}
-          name="afp"
-          defaultValue={afp}
-          onChange={handleRadioChange}
-          error={validationError}
-          role="radiogroup"
-          aria-required="true"
-        >
-          <Radio value="ja_offentlig">
-            <FormattedMessage id="stegvisning.afp.radio_ja_offentlig" />
-          </Radio>
-          {showAlert === 'ja_offentlig' &&
-            !afpOffentligFeatureToggle?.enabled && (
+    !shouldRedirectTo && (
```

Comment:
Er lettere å lese om man heller lager en `if(shouldRedirectTo) { return null }` over disse? I stedet for å pakke inn hele komponentet?

Reply:
Enig! Er det greit for deg om jeg tar det i en oppfølgingsPR?